gpaw/default.nix

```diff
@@ -0,0 +1,143 @@
+{ fetchFromGitLab, buildPythonPackage, lib, writeTextFile, makeWrapper, fetchurl
+, blas
+, lapack
+, scalapack
+, mpi
+, fftw
+, libxc
+, libvdwxc
+, which
+# Python dependencies
+, ase
+, numpy
+, scipy
+}:
+assert
+  lib.asserts.assertMsg
+  (!blas.isILP64)
+  "A 32 bit integer implementation of BLAS is required.";
+
+assert
+  lib.asserts.assertMsg
+  (!lapack.isILP64)
+  "A 32 bit integer implementation of LAPACK is required.";
+
+let
+  gpawConfig = writeTextFile {
+    name = "siteconfig.py";
+    text = ''
+      # Compiler
+      compiler = 'gcc'
+      mpicompiler = '${mpi}/bin/mpicc'
+      mpilinker = '${mpi}/bin/mpicc'
+
+      # BLAS
+      libraries += ['blas']
+      library_dirs += ['${blas}/lib']
+
+      # FFTW
+      fftw = True
+      if fftw:
+        libraries += ['fftw3']
+
+      scalapack = True
+      if scalapack:
+        libraries += ['scalapack']
+
+      # LibXC
+      libxc = True
+      if libxc:
+        xc = '${libxc}/'
+        include_dirs += [xc + 'include']
+        library_dirs += [xc + 'lib/']
+        extra_link_args += ['-Wl,-rpath={xc}/lib'.format(xc=xc)]
+        if 'xc' not in libraries:
+          libraries.append('xc')
+
+      # LibVDWXC
+      libvdwxc = True
+      if libvdwxc:
+        vdwxc = '${libvdwxc}/'
+        extra_link_args += ['-Wl,-rpath=%s/lib' % vdwxc]
+        library_dirs += ['%s/lib' % vdwxc]
+        include_dirs += ['%s/include' % vdwxc]
+        libraries += ['vdwxc']
+    '';
+  };
+
+  setupVersion = "0.9.20000";
+  pawDataSets = fetchurl {
+    url = "https://wiki.fysik.dtu.dk/gpaw-files/gpaw-setups-${setupVersion}.tar.gz";
+    sha256 = "07yldxnn38gky39fxyv3rfzag9p4lb0xfpzn15wy2h9aw4mnhwbc";
+  };
+in
+  buildPythonPackage rec {
+    pname = "gpaw";
+    version = "21.1.0";
+
+    nativeBuildInputs = [
+      which
+      makeWrapper
+    ];
+
+    buildInputs = [
+      blas
+      scalapack
+      fftw
+      libxc
+      libvdwxc
+    ];
+
+    propagatedBuildInputs = [
+      ase
+      scipy
+      numpy
+      mpi
+    ];
+
+    src = fetchFromGitLab  {
+      owner = "gpaw";
+      repo = pname;
+      rev = version;
+      sha256 = "0xws39yq5d1ih4zj31bdd0h64afs6x9j30yrrycxf3pnc5h9wbr0";
+    };
+
+    preConfigure = ''
+      export PATH=$PATH:${mpi}/bin
+    '';
+
+    postInstall = ''
+      currDir=$(pwd)
+      mkdir -p $out/share/gpaw && cd $out/share/gpaw
+      cp ${pawDataSets} gpaw-setups.tar.gz
+      tar -xvf $out/share/gpaw/gpaw-setups.tar.gz
+      rm gpaw-setups.tar.gz
+      cd $currDir
+    '';
+
+    doCheck = false;
+    preCheck = ''
+      # export PATH=$PATH:${mpi}/bin
+      export GPAW_SETUP_PATH=$out/share/gpaw/gpaw-setups-${setupVersion}
+    '';
+
+    postPatch = ''
+      cp ${gpawConfig} siteconfig.py
+    '';
+
+    postFixup = ''
+      for exe in $out/bin/*; do
+        wrapProgram $exe \
+          --set "GPAW_SETUP_PATH" "$out/share/gpaw/gpaw-setups-${setupVersion}"
+      done
+      ln -s ${mpi}/bin/mpiexec $out/bin/.
```

Comment:
I have done something like this for CP2K but I am not convinced we should keep linking `mpirun` into `$out/bin` as this could lead to collisions. Let's remove it for now and look for a better (nix compatible) way to do this.

Reply:
removed linking of MPI. On a first thought i have no idea how to provide the MPI executables nicely

Reply:
There are several ways to deal with it:
1. On an HPC cluster with properly set up PMIX one can just use `srun` instead of `mpirun`. 
2. Setting up the job with e.g. `nix-shell -p gpaw mpi` also works (little more typing though but a clean solution). 
3. We could use the `passthru` attribute:
```
passthru.mpi = mpi;
```
Then you can compose your environment like this: `nix-shell -p gpaw gpaw.passthru.mpi` (even more typing but a very clean solution).